osquery/database/database.cpp

```diff
@@ -266,10 +270,16 @@ inline void addLegacyFieldsAndDecorations(const QueryLogItem& item,
 
   // Append the decorations.
   if (item.decorations.size() > 0) {
-    tree.add_child("decorations", pt::ptree());
-    auto& decorations = tree.get_child("decorations");
-    for (const auto& name : item.decorations) {
-      decorations.put<std::string>(name.first, name.second);
+    if (FLAGS_top_level_decorators) {
+      for (const auto& name : item.decorations) {
```

Comment:
@theopolis I initially tried to do this, but went with an if/else with 2 loops because I couldn't figure out a way to do this without running into reference reassignment issues.  

So in the example above, you initially set the auto reference `decorator_parent` to tree, then when the the top level decorator flag is false, you set `decorator_parent` to a child tree.  The reassignment to the child tree ends up not working when I run this code.

Maybe my brain is not working today.  Am I looking at this wrong?  get_child returns a ptree ref and we cant reassign refs correct?